backend/oasst_backend/prompt_repository.py

```diff
@@ -488,3 +496,178 @@ def close_task(self, post_id: str, allow_personal_tasks: bool = False):
         wp.done = True
         self.db.add(wp)
         self.db.commit()
+
+    @staticmethod
+    def trace_conversation(messages: list[Post] | dict[UUID, Post], last_message: Post) -> list[Post]:
+        """
+        Pick messages from a collection so that the result makes a linear conversation
+        starting from a message tree root and up to the given message.
+        Returns an ordered list of messages starting from the message tree root.
+        """
+        if isinstance(messages, list):
+            messages = {m.id: m for m in messages}
+        if not isinstance(messages, dict):
+            # This should not normally happen
+            raise OasstError("Server error", OasstErrorCode.SERVER_ERROR, HTTPStatus.INTERNAL_SERVER_ERROR)
```

Comment:
maybe add also a unique id here, to quickly identify the error-site from the error code. The text can stay generic if we do not want to expose to the user what went wrong.

Reply:
I'd rather not, this is a type of a situation which would never happen if 1) typing is enforced in the code and controlled with linters and 2) there's sufficient validation on fastapi/pydantic level. 

How would you describe this error case? I'd say "something wrong happened and we don't really know why" :)
If you'd describe it as "we expected one type/value somewhere, but got something else", then why would you give such amount of details to the frontend (which is the ultimate reciever of these errors so far unless you read ALL the server logs manually)? What would it be for?

If you really want to control what errors arise at which line than I'd recommend look towards Sentry and/or Prometheus+Kibana/Grafana.